File: app/src/main/java/com/by5388/sy95306v2/base/adapter/FragmentAdapter.java
package com.by5388.sy95306v2.base.adapter;

import com.by5388.sy95306v2.bean.MyViewPager;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

/**
 * TODO  ViewPager 适配器的构造方法、源码、使用方法
 *
 * @author by5388
 * @date 20180727
 */
public class FragmentAdapter extends FragmentStatePagerAdapter {
    private final List<MyViewPager> viewPagers;


    public FragmentAdapter(@NonNull FragmentManager manager, @NonNull List<MyViewPager> viewPagers) {
        super(manager);
        this.viewPagers = viewPagers;
    }


    @Override
    public Fragment getItem(int position) {
        return viewPagers.get(position).getFragments();
    }

    @Override
    public int getCount() {
        return viewPagers.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return viewPagers.get(position).getTitles();
    }
}

File: sy95306v1kotlin/src/main/java/com/by5388/sy95306/kotlin/main/adapter/FragmentAdapter.kt
package com.by5388.sy95306.kotlin.main.adapter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter

/**
 * @author  by5388  on 2019/1/5.
 */
class FragmentAdapter(fragmentManager: FragmentManager, private val fragments: List<Fragment>) : FragmentStatePagerAdapter(fragmentManager) {

    override fun getItem(position: Int) = fragments[position]

    override fun getCount() = fragments.size
}